src/erlcloud_guardduty.erl

```diff
@@ -0,0 +1,144 @@
+    ok.
+
+%%
+%% API
+%%
+
+%%------------------------------------------------------------------------------
+%% @doc
+%% GuardDuty API:
+%% [http://docs.aws.amazon.com/guardduty/....html]
+%%
+%% ===Example===
+%%
+%%------------------------------------------------------------------------------
+
+-spec describe_detector(DetectorId :: binary()) -> gd_return().
+describe_detector(DetectorId) ->
+    describe_detector(DetectorId, default_config()).
+
+-spec describe_detector(DetectorId :: binary(),
+                        Config       :: aws_config()) -> gd_return().
+describe_detector(DetectorId, Config) ->
+    Path = base_path() ++ "detector/" ++ binary_to_list(DetectorId),
+    guardduty_request(Config, get, Path, undefined).
+
+
+%%------------------------------------------------------------------------------
+%% @doc
+%% GuardDuty API:
+%% [http://docs.aws.amazon.com/guardduty/....html]
+%%
+%% ===Example===
+%%
+%%------------------------------------------------------------------------------
+
+-spec list_detectors() -> gd_return().
+list_detectors() -> list_detectors(default_config()).
+
+-spec list_detectors(Config :: aws_config()) -> gd_return().
+list_detectors(Config) ->
+    list_detectors(undefined, undefined, Config).
+
+-spec list_detectors(Marker   :: binary(),
+                     MaxItems :: integer()) -> gd_return().
+list_detectors(Marker, MaxItems) ->
+    list_detectors(Marker, MaxItems, default_config()).
+
+-spec list_detectors(Marker   :: binary(),
+                     MaxItems :: integer(),
+                     Config   :: aws_config()) -> gd_return().
+list_detectors(Marker, MaxItems, Config) ->
+    Path = base_path() ++ "detector",
+    QParams = filter_undef([{"Marker", Marker},
+                            {"MaxItems", MaxItems}]),
+    guardduty_request(Config, get, Path, undefined, QParams).
+
+
+filter_undef(List) ->
+    lists:filter(fun({_Name, Value}) -> Value =/= undefined end, List).
+
+base_path() ->
+    "/".
+
+guardduty_request(Config, Method, Path, Body) ->
+    guardduty_request(Config, Method, Path, Body, []).
+
+guardduty_request(Config, Method, Path, Body, QParam) ->
+    case erlcloud_aws:update_config(Config) of
+        {ok, Config1} ->
+            guardduty_request_no_update(Config1, Method, Path, Body, QParam);
+        {error, Reason} ->
+            {error, Reason}
+    end.
+
+guardduty_request_no_update(Config, Method, Path, Body, QParam) ->
+    Form = case encode_body(Body) of
+               <<>>   -> erlcloud_http:make_query_string(QParam);
+               Value  -> Value
+           end,
+    Headers = headers(Method, Path, Config, encode_body(Body), QParam),
+    io:format("HEADERS: ~p~n", [Headers]),
+    case erlcloud_aws:aws_request_form_raw(
+           Method, Config#aws_config.guardduty_scheme, Config#aws_config.guardduty_host,
+           Config#aws_config.guardduty_port, Path, Form, Headers, Config) of
+        {ok, Data} ->
+            {ok, jsx:decode(Data)};
```

Comment:
to my memory jsx does not like `<<>>` and that deserves a separate wrapper/clause.

Reply:
But AWS requests don't return empty binaries, do they? In case an empty binary is passed to `jsx:decode` here, the call should fail, and rightfully so: it's an exceptional situation.